pkg/assembler/graphql/resolvers/artifact.resolvers.go

```diff
@@ -12,13 +12,24 @@ import (
 )
 
 // IngestArtifact is the resolver for the ingestArtifact field.
-func (r *mutationResolver) IngestArtifact(ctx context.Context, artifact *model.ArtifactInputSpec) (*model.Artifact, error) {
-	return r.Backend.IngestArtifact(ctx, artifact)
+func (r *mutationResolver) IngestArtifact(ctx context.Context, artifact *model.ArtifactInputSpec) (string, error) {
+	ingestedArtifact, err := r.Backend.IngestArtifact(ctx, artifact)
+	if err != nil {
+		return "", err
+	}
+	return ingestedArtifact.ID, err
```

Comment:
This is ok for now, but we still do a lot of work in the resolvers to retrieve the entire struct. We will need to push down and have all resolvers just return the id directly from the backend.

Could be done in separate PR(s).

Also, I think we should not merge this until we do another release, since this is breaking API compatibility.

Reply:
Would you like me to change the backend so that the resolvers can just return the ID directly?

Reply:
We can break that into other PRs and I can help you with some of the backend implementations (such as arango) as they can be optimized.

Reply:
you can start with the `inmem` ones for now.

Reply:
Given this is huge, I'd say let's merge as it is and then work on the backends.

We will need to convert the resolver for the same functionality at the same time across all backends, given that these are implemented as an interface.